src/Infra/MySQL/MySQLItemRepository.ts

```diff
@@ -100,4 +101,26 @@ export class MySQLItemRepository extends Repository<Item> implements ItemReposit
       )
       .execute()
   }
+
+  async markAsDeleted(item: Item): Promise<void> {
+    const timer = new Timer()
```

Comment:
please do not instantiate dependencies in methods - they should be injected with the `@inject` tag. 

Also it's not necesarilly needed to inject timer into the item repository. I would pass the timestamp as an argument to this function.

Reply:
Would there be any overhead by injecting `Timer` to this class? I think it makes more sense to have everything under `markAsDeleted`, instead of doing something like:

```javascript
const timer = new Timer()
const timestamp = timer.getTimestampInMicroseconds()
...
await itemRepository.markAsDeleted(item, timestamp)
```